sdk/core/azure-core/src/main/java/com/azure/core/util/metrics/DefaultMeterProvider.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.metrics;
+
+import com.azure.core.util.MetricsOptions;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.ServiceLoader;
+
+final class DefaultMeterProvider implements MeterProvider {
+    public static final MeterProvider INSTANCE = new DefaultMeterProvider();
+
+    private static MeterProvider meterProvider;
+
+    private DefaultMeterProvider() {
+    }
+
+    static {
+        // Use as classloader to load provider-configuration files and provider classes the classloader
+        // that loaded this class. In most cases this will be the System classloader.
+        // But this choice here provides additional flexibility in managed environments that control
+        // classloading differently (OSGi, Spring and others) and don't/ depend on the
+        // System classloader to load Meter classes.
+        ServiceLoader<MeterProvider> serviceLoader = ServiceLoader.load(MeterProvider.class, MeterProvider.class.getClassLoader());
+        Iterator<?> iterator = serviceLoader.iterator();
+        if (iterator.hasNext()) {
+            meterProvider = serviceLoader.iterator().next();
+        }
```

Comment:
If there are more than one `MeterProvider`s in the classpath, do we expect to use the first one that was found or do we want a more deterministic behavior here and throw if there is more than one provider? Also, add a log here to indicate that a meter provider was found and include the classname of the provider.

Reply:
good point, I don't think we should throw - at the end of the day, if the wrong metrics provider is picked up, app would still be functional. 

I think we could allow users to configure the preference via env var or sys property if we ever need it, otherwise, we don't know which one to pick. We could also allow multiple providers and broadcast everything (I doubt we'd ever need it). Both options could be added later. 

Logging - 👍

Reply:
> good point, I don't think we should throw - at the end of the day, if the wrong metrics provider is picked up, app would still be functional.

I'm not sure about this. If serious system looses telemetry that's Sev1 that prompts for fast rollback.

Reply:
But if Databricks/spark/other friends or just a complicated app includes a few extra dependencies, then app that was functioning even if it doesn't need metrics would throw and would create a SEV1. 

We also don't have any strictness on tracing or even logging - SLF4J behaves exactly as I did it on metrics - https://www.baeldung.com/slf4j-classpath-multiple-bindings. 

We do the same for HTTP client providers.

Reply:
>Would existing apps throw if we introduce this feature? I'd assume that they'd need to bring some telemetry providers on classpath first to make it happen - i.e. make intentional change to include them.

existing apps no, but assuming spring implements micrometer plugin, it'd be totally possible for customers to accidentally get two providers on the classpath. Intentional or not (through transient dependencies user has limited understanding of), it's not erroneous to have multiple metrics providers on the classpath, especially if you use none of them.

With httpclient we can never guarantee that only one httpclient provider is on the classpath - it's totally up to the combination of dependencies to bring multiple clients on the classpath.

I think what we can do is to make resolution stable - e.g. I can sort providers by name and take the first one. This way if there is a problem with multiple providers, it'd be obvious with minimal testing and easily reproducible.

Reply:
per offline discussion, decided to start with fail-fast approach and see if it introduces any issues or if there is a demand for multiple provider support.

As a follow-up PR I'll enable a configuration that allows specifying a provider when many are available on a classpath and will also investigate unit-testing for multiple impls.